client/src/favorites-provider.tsx

```diff
@@ -0,0 +1,68 @@
+import _ from 'lodash';
+import React, {useState, useEffect} from 'react';
+import {AsyncStorage} from 'react-native';
+import * as Types from './types/graphql';
+import {messageBox} from './services';
+
+export type Favorite = Types.Poi | Types.Route;
+
+type FavoritesContextType = {
+    favorites: Favorite[],
+    isFavorite: (item: Types.Poi) => boolean,
+    addFavorite: (item: Types.Poi) => void,
+    removeFavorite: (item: Types.Poi) => void,
+};
+
+type FavoritesProviderProps = {
+    children: React.ReactNode,
+};
+
+const favoritesStorageKey = 'favorites';
+
+const defaultValue: FavoritesContextType = {
+    favorites: [],
```

Comment:
thought it might better to have separate collections for routes and pois 🤔

Reply:
thought about it, too. it's easier if we take them as abstract entities. all of them have `__typename` and `id`, so we can differentiate between them when needed.